sounds/models.py

```diff
@@ -589,56 +591,34 @@ def set_license(self, new_license):
         self.license = new_license
         SoundLicenseHistory.objects.create(sound=self, license=new_license)
 
-    # N.B. These set functions are used in the distributed processing.
-    # They set a single field to prevent overwriting eachother's result in
-    # the database, which is what happens if you use Django's save() method.
-    def set_single_field(self, field, value, include_quotes=True):
-        self.set_fields([[field, value, include_quotes]])
-
-    def set_fields(self, fields):
-        query = "UPDATE sounds_sound SET "
-        query += ", ".join([('%s = %s' %
-                             (field[0], (field[1] if not field[2] else ("'%s'" % field[1])))) for field in fields])
-        query += " WHERE id = %s" % self.id
-        with transaction.atomic():
-            cursor = connection.cursor()
-            cursor.execute(query)
-
-    def set_processing_state(self, state):
-        self.set_single_field('processing_state', state)
+    # N.B. The set_xxx functions below are used in the distributed processing and other parts of the app where we only
+    # want to save an individual field of the model to prevent overwritting other model fields.
 
     def set_processing_ongoing_state(self, state):
-        self.set_single_field('processing_ongoing_state', state)
-
-    def set_processing_date(self, date):
-        self.set_single_field('processing_date', str(date))  # Date should be datetime object
+        self.processing_ongoing_state = state
+        self.save(update_fields=['processing_ongoing_state'])
 
     def set_analysis_state(self, state):
-        self.set_single_field('analysis_state', state)
+        self.analysis_state = state
+        self.save(update_fields=['analysis_state'])
 
     def set_similarity_state(self, state):
-        self.set_single_field('similarity_state', state)
-
-    def set_moderation_state(self, state):
-        self.set_single_field('moderation_state', state)
-
-    def set_moderation_date(self, date):
-        self.set_single_field('moderation_date', str(date))  # Date should be datetime object
-
-    def set_original_path(self, path):
-        self.set_single_field('original_path', path)
+        self.similarity_state = state
+        self.save(update_fields=['similarity_state'])
 
     def set_audio_info_fields(self, info):
```

Comment:
this method is only used once, so consider refactoring it to pass all fields in as arguments to make it more explicit